.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,357 @@
+              install_hole: true
+              codecov: true
+              extra_deps: "EXTRA PY35"
+            - name: macOS
+              os: macOS-latest
+              python-version: 3.7
+              run_type: FULL
+              install_hole: true
+              codecov: true
+            - name: minimal-ubuntu
+              os: ubuntu-latest
+              python-version: 3.6
+              extra_deps: ""
+              install_hole: false
+              codecov: true
+            - name: numpy_min
+              os: ubuntu-latest
+              python-version: 3.6
+              extra_deps: ""
+              install_hole: false
+              codecov: false
+              numpy: numpy=1.16.0
+            - name: asv_check
+              os: ubuntu-latest
+              python-version: 3.7
+              extra_deps: "EXTRA PY3"
+              install_hole: false
+              codecov: false
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+      GH_OS: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        # Set OS specific vars and compiler flags
+        echo "OS_NAME=osx" >> $GITHUB_ENV
+        # TODO: work out why this is necessary (from CI helpers)
+        echo "MACOSX_DEPLOYMENT_TARGET=10.9" >> $GITHUB_ENV
+        ulimit -S -n 2048
+        clang -v
+        echo "CC=clang" >> $GITHUB_ENV
+        clang++ -v
+        echo "CXX=clang++" >> $GITHUB_ENV
+        gfortran-9 -v
+        echo "FC=gfortran-9" >> $GITHUB_ENV
+
+    - name: setup_linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        # Set OS specific vars and compiler flags
+        echo "OS_NAME=linux" >> $GITHUB_ENV
+        gcc -v
+        echo "CC=gcc" >> $GITHUB_ENV
+        g++ -v
+        echo "CXX=g++" >> $GITHUB_ENV
+        gfortran -v
+        echo "FC=gfortran" >> $GITHUB_ENV
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        auto-update-conda: true
+        channel-priority: flexible
+        channels: biobuilds, conda-forge, defaults
+        add-pip-as-python-dependency: true
+        miniconda-version: ${{ matrix.conda-version }}
+        # TODO: mamba causes pip to segfault, switch when fixed
+        #mamba-version: "*"
+        architecture: x64
+        
+    - name: install_deps
+      run: |
+        # NOTE: vars need to be re-assigned
+        # NOTE: set matrix.numpy to pin to a specific numpy version
+        conda_deps="${{ matrix.numpy }} ${MDA_CONDA_MIN_DEPS}"
+        pip_deps="${MDA_PIP_MIN_DEPS}"
+
+        for dep in ${{ matrix.extra_deps }}; do
+          conda_var="MDA_CONDA_${dep}_DEPS"
+          pip_var="MDA_PIP_${dep}_DEPS"
+          echo "${!conda_var}"
+          conda_deps="$conda_deps ${!conda_var}"
+          echo $conda_deps
+          pip_deps="$pip_deps ${!pip_var}"
+        done
```

Comment:
I changed this part from @IAlibay's original setup because setting up all the environment variables was a little bit of a pain. I am however happy to go back to the original construct now that we've figured out which dependencies to list.

Reply:
Could do -- we could also have `environment.yml` files for pythons 2.7, 3.5, fully featured, and minimal. That also saves users from having to work this stuff out.

Reply:
So the reason I chose to avoid env files in `develop` is that they cause really big slowdowns for some reason, conda setup time was going from ~ 3 mins to ~ 15 mins. That and mamba 100% hates pip in environment files.

I'd prefer stay away from them on `develop`, but if we need them to clean up the matrix here, I don't have any objections (as long as CI actually completes).